cranelift/filetests/src/test_run.rs

```diff
@@ -60,7 +63,17 @@ impl SubTest for TestRun {
                 // running x86_64 code on aarch64 platforms.
                 let compiled_fn = compiler.compile(func.clone().into_owned())?;
                 command
-                    .run(|_, args| Ok(compiled_fn.call(args)))
+                    .run(|_, run_args| {
+                        let runtime_struct = test_env.runtime_struct();
+
+                        let mut args = Vec::with_capacity(run_args.len());
+                        if test_env.is_active() {
+                            args.push(runtime_struct.pointer(context.isa.unwrap().pointer_type()));
+                        }
+                        args.extend_from_slice(run_args);
+
+                        Ok(compiled_fn.call(&args))
```

Comment:
It seems like here, or somewhere, we should be checking the `Function` signature to see if its arguments begin with a `vmctx` and, if not, emit a useful error like "you need to change your function signature to..."